internal/users/tempentries/groups.go

```diff
@@ -0,0 +1,179 @@
+package tempentries
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ubuntu/authd/internal/log"
+	"github.com/ubuntu/authd/internal/users/localentries"
+	"github.com/ubuntu/authd/internal/users/types"
+)
+
+type groupRecord struct {
+	name   string
+	gid    uint32
+	passwd string
+}
+
+type temporaryGroupRecords struct {
+	idGenerator   IDGenerator
+	registerMutex sync.Mutex
+	rwMutex       sync.RWMutex
+	groups        map[uint32]groupRecord
+	gidByName     map[string]uint32
+}
+
+func newTemporaryGroupRecords(idGenerator IDGenerator) *temporaryGroupRecords {
+	return &temporaryGroupRecords{
+		idGenerator:   idGenerator,
+		registerMutex: sync.Mutex{},
+		rwMutex:       sync.RWMutex{},
+		groups:        make(map[uint32]groupRecord),
+		gidByName:     make(map[string]uint32),
+	}
+}
+
+// GroupByID returns the group information for the given group ID.
+func (r *temporaryGroupRecords) GroupByID(gid uint32) (types.GroupEntry, error) {
+	r.rwMutex.RLock()
+	defer r.rwMutex.RUnlock()
+
+	group, ok := r.groups[gid]
+	if !ok {
+		return types.GroupEntry{}, NoDataFoundError{}
+	}
+
+	return r.groupEntry(group), nil
+}
+
+// GroupByName returns the group information for the given group name.
+func (r *temporaryGroupRecords) GroupByName(name string) (types.GroupEntry, error) {
+	r.rwMutex.RLock()
+	defer r.rwMutex.RUnlock()
+
+	gid, ok := r.gidByName[name]
+	if !ok {
+		return types.GroupEntry{}, NoDataFoundError{}
+	}
+
+	return r.GroupByID(gid)
+}
+
+func (r *temporaryGroupRecords) groupEntry(group groupRecord) types.GroupEntry {
+	return types.GroupEntry{Name: group.name, GID: group.gid, Passwd: group.passwd}
+}
```

Comment:
Does this need to be part of `temporaryGroupRecords`?

Reply:
done